src/Configuration/Schema/SchemaConfiguration.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Configuration\Schema;
+
+use Infection\Configuration\Entry\Logs;
+use Infection\Configuration\Entry\PhpUnit;
+use Infection\Configuration\Entry\Source;
+use Webmozart\Assert\Assert;
+
+/**
+ * @internal
+ */
+final class SchemaConfiguration
+{
+    private const TEST_FRAMEWORKS = [
+        'phpunit',
+        'phpspec',
+    ];
+
+    private $file;
+    private $timeout;
+    private $source;
+    private $logs;
+    private $tmpDir;
+    private $phpUnit;
+    private $mutators;
+    private $testFramework;
+    private $bootstrap;
+    private $initialTestsPhpOptions;
+    private $testFrameworkOptions;
+
+    public function __construct(
+        string $file,
+        ?int $timeout,
+        Source $source,
+        Logs $logs,
+        ?string $tmpDir,
+        PhpUnit $phpUnit,
+        array $mutators,
+        ?string $testFramework,
+        ?string $bootstrap,
+        ?string $initialTestsPhpOptions,
+        ?string $testFrameworkOptions
+    ) {
+        Assert::nullOrGreaterThanEq($timeout, 1);
+        Assert::nullOrOneOf($testFramework, self::TEST_FRAMEWORKS);
```

Comment:
same comment as before: those guards are the last resort to prevent the system to run with invalid data. This however should be avoided, here for example by validation rules with the `schema.json`